types/src/content-working-group/index.ts

```diff
@@ -494,87 +290,52 @@ export type ILead = {
   inducted: BlockNumber
   stage: LeadRoleState
 }
-export class Lead extends JoyStruct<ILead> {
-  constructor(value?: ILead) {
-    super(
-      {
-        member_id: MemberId,
-        role_account: GenericAccountId,
-        reward_relationship: Option.with(RewardRelationshipId),
-        inducted: u32,
-        stage: LeadRoleState,
-      },
-      value
-    )
-  }
-
-  get member_id(): MemberId {
-    return this.getField<MemberId>('member_id')
-  }
-
-  get role_account(): GenericAccountId {
-    return this.getField<GenericAccountId>('role_account')
-  }
-
+export class Lead
+  extends JoyStructDecorated({
+    member_id: MemberId,
+    role_account: AccountId,
+    reward_relationship: Option.with(RewardRelationshipId),
+    inducted: u32,
+    stage: LeadRoleState,
+  })
+  implements ILead {
   // Helper for working-group compatibility
-  get role_account_id(): GenericAccountId {
+  get role_account_id(): AccountId {
     return this.role_account
   }
-
-  get reward_relationship(): Option<RewardRelationshipId> {
-    return this.getField<Option<RewardRelationshipId>>('reward_relationship')
-  }
-
-  get stage(): LeadRoleState {
-    return this.getField('stage')
-  }
-}
-
-export class WorkingGroupUnstaker extends Enum {
-  constructor(value?: any, index?: number) {
-    super(
-      {
-        Lead: LeadId,
-        Curator: CuratorId,
-      },
-      value,
-      index
-    )
-  }
 }
 
-export class CuratorApplicationIdToCuratorIdMap extends BTreeMap<ApplicationId, CuratorId> {
-  constructor(value?: any) {
-    super(ApplicationId, CuratorId, value)
-  }
-}
+export class WorkingGroupUnstaker extends JoyEnum({
+  Lead: LeadId,
+  Curator: CuratorId,
+}) {}
 
-export function registerContentWorkingGroupTypes() {
-  try {
-    getTypeRegistry().register({
-      ChannelId: 'u64',
-      CuratorId: 'u64',
-      CuratorOpeningId: 'u64',
-      CuratorApplicationId: 'u64',
-      LeadId: 'u64',
-      PrincipalId: 'u64',
-      OptionalText,
-      Channel,
-      ChannelContentType,
-      ChannelCurationStatus,
-      ChannelPublicationStatus,
-      CurationActor,
-      Curator,
-      CuratorApplication,
-      CuratorOpening,
-      Lead,
-      OpeningPolicyCommitment,
-      Principal,
-      WorkingGroupUnstaker,
-      CuratorApplicationIdToCuratorIdMap,
-      CuratorApplicationIdSet: Vec.with(CuratorApplicationId),
-    })
-  } catch (err) {
-    console.error('Failed to register custom types of content working group module', err)
-  }
-}
+export class CuratorApplicationIdToCuratorIdMap extends BTreeMap.with(ApplicationId, CuratorId) {}
+
+export class CuratorApplicationIdSet extends Vec.with(CuratorApplicationId) {}
+
+export const contentWorkingGroupTypes = {
+  ChannelId: 'u64',
```

Comment:
It seems unnecessary to specify `ChannelId: 'u64'` and for the remaining Ids, but this is an artifact from previous version. I can't remember why I was forced to use this style however.